### PR TITLE
Custom build pipeline

### PR DESCRIPTION
**Pull request by Kink3d**
- Build to an array of targets (platform and API)
- Auto switches focus and sets up build for each
- Added build configuration files to handle this
- Cleanup some editor UI

Also includes:
ARP API fixes

### CODE REVIEW

File: Core/Build/Editor/BuildPipeline.cs [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+                    applicationName += ".app";
+                    break;
+                case UnityEditor.BuildTarget.StandaloneLinux:
+                    applicationName += ".app";
+                    break;
+                case UnityEditor.BuildTarget.StandaloneLinux64:
+                    applicationName += ".app";
+                    break;
+                case UnityEditor.BuildTarget.iOS:
+                    applicationName += ".app";
```
Comment (Verasl) [inline Core/Build/Editor/BuildPipeline.cs]: This probably shouldn't be ".app" as we will be getting an Xcode project

File: Core/Build/Editor/BuildPipeline.cs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+                        Debug.LogError(build); // Return build fail log
+                    }
+                }
+            }
+            else // Null build configuartion
+                Debug.LogError("No Build Configuration file assigned. Please assign a Build Configuration file i nthe Build Pipeline window"); // Log error 
+        }
+
+        // Try to set Graphics API for the target
+        public static bool SetGraphicsAPI(BuildTarget target)
```
Comment (Verasl) [inline Core/Build/Editor/BuildPipeline.cs]: I think we should append the Gfx API to the name of the app if we don't already, and for android/iOS we need to also append it to the product name so we can install multiple versions on the same device

File: Core/Build/Editor/BuildPipeline.cs (hunk 1 above)
Comment (Kink3d) [reply Core/Build/Editor/BuildPipeline.cs]: .xcodeproj

Comment (Kink3d) [reply Core/Build/Editor/BuildPipeline.cs]: Or maybe its just a folder...

**Pull request closed without merging**